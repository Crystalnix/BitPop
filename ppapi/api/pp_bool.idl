/* Copyright (c) 2011 The Chromium Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

/* This file defines the PP_Bool enumeration for use in PPAPI C headers. */

/* The PP_Bool enum is a boolean value for use in PPAPI C headers.  The
 * standard bool type is not available to pre-C99 compilers, and is not
 * guaranteed to be compatible between C and C++, whereas the PPAPI C
 * headers can be included from C or C++ code.
 */
enum PP_Bool {
  PP_FALSE = 0,
  PP_TRUE  = 1
};
