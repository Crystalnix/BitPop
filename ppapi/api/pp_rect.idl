/* Copyright (c) 2011 The Chromium Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

/* This file defines the APIs for creating a 2-dimensional rectangle. */

/* The PP_Rect struct contains the size and location of a 2D rectangle. */
struct PP_Rect {
  /* The x and y coordinates of the upper-left corner of the rectangle. */
  PP_Point point;
  /* The width and height of the rectangle. */
  PP_Size size;
};
