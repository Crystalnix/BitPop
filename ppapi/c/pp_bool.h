/* Copyright (c) 2010 The Chromium Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */
#ifndef PPAPI_C_PP_BOOL_H_
#define PPAPI_C_PP_BOOL_H_

#include "ppapi/c/pp_macros.h"

/**
 * @file
 * This file defines the PP_Bool enumeration for use in PPAPI C headers.
 */

/**
 *
 * @addtogroup Enums
 * @{
 */

/**
 * The PP_Bool enum is a boolean value for use in PPAPI C headers.  The
 * standard bool type is not available to pre-C99 compilers, and is not
 * guaranteed to be compatible between C and C++, whereas the PPAPI C
 * headers can be included from C or C++ code.
 */
typedef enum {
  PP_FALSE = 0,
  PP_TRUE = 1
} PP_Bool;
PP_COMPILE_ASSERT_SIZE_IN_BYTES(PP_Bool, 4);
/**
 * @}
 */

#endif  /* PPAPI_C_PP_BOOL_H_ */

