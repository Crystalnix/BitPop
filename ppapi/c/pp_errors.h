/* Copyright (c) 2011 The Chromium Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */
#ifndef PPAPI_C_PP_ERRORS_H_
#define PPAPI_C_PP_ERRORS_H_

/**
 * @file
 * This file defines an enumeration of all PPAPI error codes.
 */

/**
 *
 * @addtogroup Enums
 * @{
 */

/** This enumeration contains enumerators of all PPAPI error codes.
 * Errors are negative valued.
 */
enum {
  /**
   * This value is returned by a function on successful synchronous completion
   * or is passed as a result to a PP_CompletionCallback_Func on successful
   * asynchronous completion.
   */
  PP_OK = 0,

  /**
   * This value is returned by a function that accepts a PP_CompletionCallback
   * and cannot complete synchronously.  This code indicates that the given
   * callback will be asynchronously notified of the final result once it is
   * available.
   */
  PP_OK_COMPLETIONPENDING = -1,
  PP_ERROR_WOULDBLOCK = -1,  /* DEPRECATED: DO NOT USE */

  /** This value indicates failure for unspecified reasons. */
  PP_ERROR_FAILED = -2,

  /**
   * This value indicates failure due to an asynchronous operation being
   * interrupted, typically as a result of user action.
   */
  PP_ERROR_ABORTED = -3,

  /** This value indicates failure due to an invalid argument. */
  PP_ERROR_BADARGUMENT = -4,

  /** This value indicates failure due to an invalid PP_Resource. */
  PP_ERROR_BADRESOURCE = -5,

  /** This value indicates failure due to an unavailable PPAPI interface. */
  PP_ERROR_NOINTERFACE = -6,

  /** This value indicates failure due to insufficient privileges. */
  PP_ERROR_NOACCESS = -7,

  /** This value indicates failure due to insufficient memory. */
  PP_ERROR_NOMEMORY = -8,

  /** This value indicates failure due to insufficient storage space. */
  PP_ERROR_NOSPACE = -9,

  /** This value indicates failure due to insufficient storage quota. */
  PP_ERROR_NOQUOTA = -10,

  /**
   * This value indicates failure due to an action already being in
   * progress.
   */
  PP_ERROR_INPROGRESS = -11,

  /** This value indicates failure due to a file that does not exist. */
  PP_ERROR_FILENOTFOUND = -20,

  /** This value indicates failure due to a file that already exists. */
  PP_ERROR_FILEEXISTS = -21,

  /** This value indicates failure due to a file that is too big. */
  PP_ERROR_FILETOOBIG = -22,

  /**
   * This value indicates failure due to a file having been modified
   * unexpectedly.
   */
  PP_ERROR_FILECHANGED = -23,

  /** This value indicates failure due to a time limit being exceeded. */
  PP_ERROR_TIMEDOUT = -30,

  /** This value indicates that the user cancelled rather than providing
   * expected input.
   */
  PP_ERROR_USERCANCEL = -40
};

/**
 * @}
 * End of addtogroup Enums
 */

#endif  /* PPAPI_C_PP_ERRORS_H_ */
