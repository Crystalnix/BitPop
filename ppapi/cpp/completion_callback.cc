// Copyright (c) 2011 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "ppapi/cpp/completion_callback.h"

namespace pp {

CompletionCallback BlockUntilComplete() {
  return CompletionCallback();
}

}  // namespace pp
