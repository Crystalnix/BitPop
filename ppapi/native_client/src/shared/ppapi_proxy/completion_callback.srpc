# Copyright (c) 2010 The Native Client Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

# RPC method that supports invocation of plugin-side callbacks from the browser.
{
 'name': 'CompletionCallbackRpc',
 'rpcs': [
          {'name': 'RunCompletionCallback',
           'inputs': [['callback_id', 'int32_t'],
                      ['result', 'int32_t'],
                      ['read_buffer', 'char[]' ],
                     ],
           'outputs': []
          },
         ]
}
