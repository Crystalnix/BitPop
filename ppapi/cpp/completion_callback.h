// Copyright (c) 2011 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#ifndef PPAPI_CPP_COMPLETION_CALLBACK_H_
#define PPAPI_CPP_COMPLETION_CALLBACK_H_

#include "ppapi/c/pp_completion_callback.h"
#include "ppapi/c/pp_errors.h"
#include "ppapi/cpp/logging.h"
#include "ppapi/cpp/module.h"

/// @file
/// This file defines the API to create and run a callback.
namespace pp {

/// This API enables you to implement and receive callbacks when
/// Pepper operations complete asynchronously.
class CompletionCallback {
 public:
  /// The default constructor will create a blocking
  /// <code>CompletionCallback</code> that can be passed to a method to
  /// indicate that the calling thread should be blocked until the asynchronous
  /// operation corresponding to the method completes.
  ///
  /// <strong>Note:</strong> Blocking completion callbacks are only allowed from
  /// from background threads.
  CompletionCallback() {
    cc_ = PP_BlockUntilComplete();
  }

  /// A constructor for creating a <code>CompletionCallback</code>.
  ///
  /// @param[in] func The function to be called on completion.
  /// @param[in] user_data The user data to be passed to the callback function.
  /// This is optional and is typically used to help track state in case of
  /// multiple pending callbacks.
  CompletionCallback(PP_CompletionCallback_Func func, void* user_data) {
    cc_ = PP_MakeCompletionCallback(func, user_data);
  }

  /// A constructor for creating a <code>CompletionCallback</code> with
  /// specified flags.
  ///
  /// @param[in] func The function to be called on completion.
  /// @param[in] user_data The user data to be passed to the callback function.
  /// This is optional and is typically used to help track state in case of
  /// multiple pending callbacks.
  /// @param[in] flags Bit field combination of
  /// <code>PP_CompletionCallback_Flag</code> flags used to control how
  /// non-NULL callbacks are scheduled by asynchronous methods.
  CompletionCallback(PP_CompletionCallback_Func func, void* user_data,
                     int32_t flags) {
    cc_ = PP_MakeCompletionCallback(func, user_data);
    cc_.flags = flags;
  }

  /// The set_flags() function is used to set the flags used to control
  /// how non-NULL callbacks are scheduled by asynchronous methods.
  ///
  /// @param[in] flags Bit field combination of
  /// <code>PP_CompletionCallback_Flag</code> flags used to control how
  /// non-NULL callbacks are scheduled by asynchronous methods.
  void set_flags(int32_t flags) { cc_.flags = flags; }

  /// Run() is used to run the <code>CompletionCallback</code>.
  /// Normally, the system runs a <code>CompletionCallback</code> after an
  /// asynchronous operation completes, but programs may wish to run the
  /// <code>CompletionCallback</code> manually in order to reuse the same code
  /// paths.
  ///
  /// @param[in] result The result of the operation to be passed to the
  /// callback function. Non-positive values correspond to the error codes
  /// from <code>pp_errors.h</code> (excluding
  /// <code>PP_OK_COMPLETIONPENDING</code>). Positive values indicate
  /// additional information such as bytes read.
  void Run(int32_t result) {
    PP_DCHECK(cc_.func);
    PP_RunCompletionCallback(&cc_, result);
  }

  /// IsOptional() is used to determine the setting of the
  /// <code>PP_COMPLETIONCALLBACK_FLAG_OPTIONAL</code> flag. This flag allows
  /// any method taking such callback to complete synchronously
  /// and not call the callback if the operation would not block. This is useful
  /// when performance is an issue, and the operation bandwidth should not be
  /// limited to the processing speed of the message loop.
  ///
  /// On synchronous method completion, the completion result will be returned
  /// by the method itself. Otherwise, the method will return
  /// PP_OK_COMPLETIONPENDING, and the callback will be invoked asynchronously
  /// on the main thread of Pepper execution.
  ///
  /// @return true if this callback is optional, otherwise false.
  bool IsOptional() const {
    return (cc_.func == NULL ||
            (cc_.flags & PP_COMPLETIONCALLBACK_FLAG_OPTIONAL) != 0);
  }

  /// The pp_completion_callback() function returns the underlying
  /// <code>PP_CompletionCallback</code>
  ///
  /// @return A <code>PP_CompletionCallback</code>.
  const PP_CompletionCallback& pp_completion_callback() const { return cc_; }

  /// The flags() function returns flags used to control how non-NULL callbacks
  /// are scheduled by asynchronous methods.
  ///
  /// @return An int32_t containing a bit field combination of
  /// <code>PP_CompletionCallback_Flag</code> flags.
  int32_t flags() const { return cc_.flags; }

  /// MayForce() is used when implementing functions taking callbacks.
  /// If the callback is required and <code>result</code> indicates that it has
  /// not been scheduled, it will be forced on the main thread.
  ///
  /// <strong>Example:</strong>
  ///
  /// @code
  ///
  /// int32_t OpenURL(pp::URLLoader* loader,
  ///                 pp::URLRequestInfo* url_request_info,
  ///                 const CompletionCallback& cc) {
  ///   if (loader == NULL || url_request_info == NULL)
  ///     return cc.MayForce(PP_ERROR_BADRESOURCE);
  ///   return loader->Open(*loader, *url_request_info, cc);
  /// }
  ///
  /// @endcode
  ///
  /// @param[in] result PP_OK_COMPLETIONPENDING or the result of the completed
  /// operation to be passed to the callback function. PP_OK_COMPLETIONPENDING
  /// indicates that the callback has already been scheduled. Other
  /// non-positive values correspond to error codes from
  /// <code>pp_errors.h</code>. Positive values indicate additional information
  /// such as bytes read.
  ///
  /// @return <code>PP_OK_COMPLETIONPENDING</code> if the callback has been
  /// forced, result parameter otherwise.
  int32_t MayForce(int32_t result) const {
    if (result == PP_OK_COMPLETIONPENDING || IsOptional())
      return result;
    Module::Get()->core()->CallOnMainThread(0, *this, result);
    return PP_OK_COMPLETIONPENDING;
  }

 protected:
  PP_CompletionCallback cc_;
};

/// BlockUntilComplete() is used in place of an actual completion callback
/// to request blocking behavior. If specified, the calling thread will block
/// until the function completes. Blocking completion callbacks are only
/// allowed from background threads.
///
/// @return A <code>CompletionCallback</code> corresponding to a NULL callback.
CompletionCallback BlockUntilComplete();

}  // namespace pp

#endif  // PPAPI_CPP_COMPLETION_CALLBACK_H_
