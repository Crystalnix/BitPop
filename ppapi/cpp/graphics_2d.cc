// Copyright (c) 2012 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "ppapi/cpp/graphics_2d.h"

#include "ppapi/c/pp_errors.h"
#include "ppapi/c/ppb_graphics_2d.h"
#include "ppapi/cpp/completion_callback.h"
#include "ppapi/cpp/image_data.h"
#include "ppapi/cpp/instance_handle.h"
#include "ppapi/cpp/module.h"
#include "ppapi/cpp/module_impl.h"
#include "ppapi/cpp/point.h"
#include "ppapi/cpp/rect.h"

namespace pp {

namespace {

template <> const char* interface_name<PPB_Graphics2D_1_0>() {
  return PPB_GRAPHICS_2D_INTERFACE_1_0;
}

}  // namespace

Graphics2D::Graphics2D() : Resource() {
}

Graphics2D::Graphics2D(const Graphics2D& other)
    : Resource(other),
      size_(other.size_) {
}

Graphics2D::Graphics2D(const InstanceHandle& instance,
                       const Size& size,
                       bool is_always_opaque)
    : Resource() {
  if (!has_interface<PPB_Graphics2D_1_0>())
    return;
  PassRefFromConstructor(get_interface<PPB_Graphics2D_1_0>()->Create(
      instance.pp_instance(),
      &size.pp_size(),
      PP_FromBool(is_always_opaque)));
  if (!is_null()) {
    // Only save the size if allocation succeeded.
    size_ = size;
  }
}

Graphics2D::~Graphics2D() {
}

Graphics2D& Graphics2D::operator=(const Graphics2D& other) {
  Resource::operator=(other);
  size_ = other.size_;
  return *this;
}

void Graphics2D::PaintImageData(const ImageData& image,
                                const Point& top_left) {
  if (!has_interface<PPB_Graphics2D_1_0>())
    return;
  get_interface<PPB_Graphics2D_1_0>()->PaintImageData(pp_resource(),
                                                  image.pp_resource(),
                                                  &top_left.pp_point(),
                                                  NULL);
}

void Graphics2D::PaintImageData(const ImageData& image,
                                const Point& top_left,
                                const Rect& src_rect) {
  if (!has_interface<PPB_Graphics2D_1_0>())
    return;
  get_interface<PPB_Graphics2D_1_0>()->PaintImageData(pp_resource(),
                                                  image.pp_resource(),
                                                  &top_left.pp_point(),
                                                  &src_rect.pp_rect());
}

void Graphics2D::Scroll(const Rect& clip, const Point& amount) {
  if (!has_interface<PPB_Graphics2D_1_0>())
    return;
  get_interface<PPB_Graphics2D_1_0>()->Scroll(pp_resource(),
                                          &clip.pp_rect(),
                                          &amount.pp_point());
}

void Graphics2D::ReplaceContents(ImageData* image) {
  if (!has_interface<PPB_Graphics2D_1_0>())
    return;
  get_interface<PPB_Graphics2D_1_0>()->ReplaceContents(pp_resource(),
                                                   image->pp_resource());

  // On success, reset the image data. This is to help prevent people
  // from continuing to use the resource which will result in artifacts.
  *image = ImageData();
}

int32_t Graphics2D::Flush(const CompletionCallback& cc) {
  if (!has_interface<PPB_Graphics2D_1_0>())
    return cc.MayForce(PP_ERROR_NOINTERFACE);
  return get_interface<PPB_Graphics2D_1_0>()->Flush(pp_resource(),
                                                cc.pp_completion_callback());
}

}  // namespace pp
