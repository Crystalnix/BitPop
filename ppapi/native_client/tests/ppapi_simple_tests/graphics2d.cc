// Copyright (c) 2011 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//

#include <stdint.h>

#include <cmath>
#include <limits>
#include <string>

#include "native_client/src/shared/platform/nacl_check.h"
#include "native_client/src/shared/platform/nacl_log.h"

#include "ppapi/c/pp_rect.h"
#include "ppapi/c/pp_bool.h"
#include "ppapi/c/pp_errors.h"
#include "ppapi/cpp/rect.h"
#include "ppapi/cpp/graphics_2d.h"
#include "ppapi/cpp/image_data.h"
#include "ppapi/cpp/completion_callback.h"
#include "ppapi/cpp/instance.h"
#include "ppapi/cpp/module.h"

const uint32_t kDefaultColor = 0x2266aa;

void FillRect(pp::ImageData* image, int left, int top, int width, int height,
              uint32_t color) {
  for (int y = std::max(0, top);
       y < std::min(image->size().height() - 1, top + height);
       y++) {
    for (int x = std::max(0, left);
         x < std::min(image->size().width() - 1, left + width);
         x++)
      *image->GetAddr32(pp::Point(x, y)) = color;
  }
}

class MyInstance : public pp::Instance {
 private:
  uint32_t color_;
  pp::Size size_;
  pp::Graphics2D device_context_;

  void ParseArgs(uint32_t argc, const char* argn[], const char* argv[]) {
    for (uint32_t i = 0; i < argc; ++i) {
      const std::string tag = argn[i];
      if (tag == "color") color_ = strtol(argv[i], 0, 0);
    }
  }

  static void FlushCallback(void* thiz, int32_t result) {
    // If necessary we can get at the instabce like so:
    // MyInstance* instance = static_cast<MyInstance*>(thiz);
  }

 public:
  explicit MyInstance(PP_Instance instance)
    : pp::Instance(instance), color_(kDefaultColor), size_(0, 0) {}

  virtual bool Init(uint32_t argc, const char* argn[], const char* argv[]) {
    ParseArgs(argc, argn, argv);
    return true;
  }

  virtual void DidChangeView(const pp::Rect& position, const pp::Rect& clip) {
    if (position.size().width() == size_.width() &&
        position.size().height() == size_.height()) {
      // No change. We don't care about the position, only the size.
      return;
    }

    device_context_ = pp::Graphics2D(this, position.size(), false);
    CHECK(BindGraphics(device_context_));
    size_ = position.size();

    pp::ImageData image(this, PP_IMAGEDATAFORMAT_BGRA_PREMUL, size_, true);
    CHECK(!image.is_null());
    FillRect(&image, 0, 0, size_.width(), size_.height(), color_);

    device_context_.ReplaceContents(&image);
    pp::CompletionCallback cc(FlushCallback, this);

    device_context_.Flush(cc);
  }
};


class MyModule : public pp::Module {
 public:
  // Override CreateInstance to create your customized Instance object.
  virtual pp::Instance* CreateInstance(PP_Instance instance) {
    return new MyInstance(instance);
  }
};

namespace pp {

// Factory function for your specialization of the Module object.
Module* CreateModule() {
  return new MyModule();
}

}  // namespace pp
