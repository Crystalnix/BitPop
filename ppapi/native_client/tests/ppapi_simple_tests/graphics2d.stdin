# To be used with
#    sel_universal_ppapi_replay_prolog.stdin
# define a bunch of variables for readability

set_variable MODULE i(4444)
set_variable INSTANCE i(5555)
# PP_Size
set_variable IMG_FMT  R(8,4:400,4:400)
set_variable IMG_HANDLE i(7)
set_variable GRAPHICS_HANLDE i(8)
# PP_ImageDataDesc
set_variable IMG_FMT2 R(16,4:0,4:400,4:400,4:1600)
# image size is width x height x 4-byte-pixel
set_variable IMG_SIZE 640000
# PP_Size
set_variable GRAPHICS_FMT  R(8,4:400,4:400)

echo
echo "*** INIT MODULE"
replay 1 PPB_GetInterface:s:i  s("PPB_Core;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_Instance;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_ImageData;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_Graphics2D;1.0") * i(1)
# The results contain a process id which is non-determinisitic. The
# 'hide-results' option suppresses this output to help with golden file.
rpc PPP_InitializeModule hide-results i(0) ${MODULE} h(pepper_desc) s("${service_string}") * i(0) i(0)

echo
echo "*** INIT INSTANCE"
set_variable TAGS C(6,color\000)
set_variable VALUES C(5,0xaa\000)

# The PPP_GetInterface() call is probably not necessary, we just emulate what
# happens in real life
rpc PPP_GetInterface s("PPP_Instance;1.0") * i(0)
rpc PPP_Instance_DidCreate ${INSTANCE} i(1) ${TAGS} ${VALUES} * i(0)

echo
echo "*** SETTTING UP VIDEO MEMORY"

# video size is image size rounded up to 64k
shmem video_handle video_address 655360
memset ${video_address} 0  ${IMG_SIZE} 0
checksum ${video_address} 0 ${IMG_SIZE}

echo
echo "*** TRIGGER REPAINT"
replay 1 PPB_Core_AddRefResource:i: ${GRAPHICS_HANLDE} *
replay 1 PPB_ImageData_Create:iiCi:i ${INSTANCE} i(0) ${IMG_FMT} i(1) * ${IMG_HANDLE}
replay 1 PPB_ImageData_Describe:i:Chii ${IMG_HANDLE} * ${IMG_FMT2} h(video_handle) i(${IMG_SIZE}) i(1)
replay 1 PPB_Graphics2D_Create:iCi:i ${INSTANCE} ${GRAPHICS_FMT} i(0) * ${GRAPHICS_HANLDE}
replay 1 PPB_Instance_BindGraphics:ii:i ${INSTANCE} ${GRAPHICS_HANLDE} * i(1)
replay 1 PPB_Graphics2D_ReplaceContents:ii:  ${GRAPHICS_HANLDE}  ${IMG_HANDLE} *
replay 1 PPB_Graphics2D_Flush:ii:i ${GRAPHICS_HANLDE} i(1) * i(-1)
replay 1 ReleaseResourceMultipleTimes:ii:  ${IMG_HANDLE} i(1) *
replay 1 PPB_Core_ReleaseResource:i:  ${GRAPHICS_HANLDE} *
rpc PPP_Instance_DidChangeView ${INSTANCE} I(4,8,79,400,400) I(4,0,0,400,400) *

echo
echo "*** CHECKING MEMORY"
checksum ${video_address} 0 ${IMG_SIZE}

rpc PPP_ShutdownModule *
