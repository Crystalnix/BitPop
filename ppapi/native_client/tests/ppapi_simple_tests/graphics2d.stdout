
"*** REPLAY PROLOG"

"*** REGISTER REPLAY METHODS"

"*** CREATE UPCALL SERVICE"

"*** INIT MODULE"
rpc call intiated PPP_InitializeModule:iihs:ii
input 0:  i(0)
input 1:  i(4444)
input 2:  h(pepper_desc)
input 3:  @LENGTHY_SERVICE_STRING
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Core;1.0")
output 0:  i(1)
rpc call complete PPP_InitializeModule:iihs:ii


"*** INIT INSTANCE"
rpc call intiated PPP_GetInterface:s:i
input 0:  s("PPP_Instance;1.0")
rpc call complete PPP_GetInterface:s:i
output 0:  i(1)

rpc call intiated PPP_Instance_DidCreate:iiCC:i
input 0:  i(5555)
input 1:  i(1)
input 2:  C(6,color\x00)
input 3:  C(5,0xaa\x00)
rpc call complete PPP_Instance_DidCreate:iiCC:i
output 0:  i(1)


"*** SETTTING UP VIDEO MEMORY"
CHECKSUM: 0x00000000

"*** TRIGGER REPAINT"
rpc call intiated PPP_Instance_DidChangeView:iII:
input 0:  i(5555)
input 1:  I(4,8,79,400,400)
input 2:  I(4,0,0,400,400)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Graphics2D;1.0")
output 0:  i(1)
replaying PPB_Graphics2D_Create:iCi:i:
input 0:  i(5555)
input 1:  C(8,\x90\x01\x00\x00\x90\x01\x00\x00)
input 2:  i(0)
output 0:  i(8)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Instance;1.0")
output 0:  i(1)
replaying PPB_Instance_BindGraphics:ii:i:
input 0:  i(5555)
input 1:  i(8)
output 0:  i(1)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_ImageData;1.0")
output 0:  i(1)
replaying PPB_ImageData_Create:iiCi:i:
input 0:  i(5555)
input 1:  i(0)
input 2:  C(8,\x90\x01\x00\x00\x90\x01\x00\x00)
input 3:  i(1)
output 0:  i(7)
replaying PPB_ImageData_Describe:i:Chii:
input 0:  i(7)
output 0:  C(16,\x00\x00\x00\x00\x90\x01\x00\x00\x90\x01\x00\x00@\x06\x00\x00)
output 1:  h(video_handle)
output 2:  i(640000)
output 3:  i(1)
replaying PPB_Graphics2D_ReplaceContents:ii::
input 0:  i(8)
input 1:  i(7)
replaying ReleaseResourceMultipleTimes:ii::
input 0:  i(7)
input 1:  i(1)
replaying PPB_Graphics2D_Flush:ii:i:
input 0:  i(8)
input 1:  i(1)
output 0:  i(-1)
rpc call complete PPP_Instance_DidChangeView:iII:


"*** CHECKING MEMORY"
CHECKSUM: 0xe1913cee
rpc call intiated PPP_ShutdownModule::
rpc call complete PPP_ShutdownModule::

